src/Orchard.Web/Modules/Orchard.OpenId/Controllers/AccessController.cs

```diff
@@ -0,0 +1,117 @@
+﻿using AspNet.Security.OpenIdConnect.Extensions;
+using AspNet.Security.OpenIdConnect.Server;
+using Microsoft.AspNetCore.Authentication;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Http.Authentication;
+using Microsoft.AspNetCore.Identity;
+using Microsoft.AspNetCore.Mvc;
+using Orchard.OpenId.Models;
+using Orchard.OpenId.ViewModels;
+using Orchard.Users.Models;
+using System.Security.Claims;
+using System.Threading.Tasks;
+
+namespace Orchard.OpenId.Controllers
+{
+    [Authorize]
+    public class AccessController : Controller
+    {
+        private readonly OpenIddict.OpenIddictApplicationManager<OpenIdApplication> _applicationManager;
+        private readonly SignInManager<User> _signInManager;
+        private readonly OpenIddict.OpenIddictUserManager<User> _userManager;
+
+        public AccessController(
+            OpenIddict.OpenIddictApplicationManager<OpenIdApplication> applicationManager,
+            SignInManager<User> signInManager,
+            OpenIddict.OpenIddictUserManager<User> userManager)
+        {
+            _applicationManager = applicationManager;
+            _signInManager = signInManager;
+            _userManager = userManager;
+        }
+
+        [Authorize, HttpGet]
```

Comment:
Finally, is it clear if [Authorize] should work as in previous versions or do we have to use the alternativa base on code? Currently authorization code flow fails because Authorize is ignored. https://github.com/OrchardCMS/Orchard2/issues/217